### DISCUSSION TITLE
Make mysql database connection parameters customizable

### DISCUSSION THREAD

**Discussion by kcpants** [hassio-addons/addon-nginx-proxy-manager#466] [Ideas]
Making the database connection parameters used by this addon customizable would allow users to run the database on another machine. This helps limit writes to the data disk which extends the lifetime of the storage. This is especially useful/important when running hassOS off of an SD card (e.g. on a Raspberry Pi without an SSD).

It looks like this might be as simple as adding optional config options and preferring them over the values in the code below if they are set (plus potentially updating/adding error messages to instruct users to check their db instance if connection attempts fail).

https://github.com/hassio-addons/addon-nginx-proxy-manager/blob/dab35df9a3b6f2df832d89d0548243b67a95c748/proxy-manager/rootfs/etc/cont-init.d/npm.sh#L130-L134

Something like:

```
# Set up database connection 
if bashio::config.has_value "mysql_host"; then mysql_host=$(bashio::config 'mysql_host'); else mysql_host=$(bashio::services "mysql" "host"); fi 
if bashio::config.has_value "mysql_password"; then mysql_password=$(bashio::config 'mysql_password'); else mysql_password=$(bashio::services "mysql" "password") ; fi 
if bashio::config.has_value "mysql_port"; then mysql_port=$(bashio::config 'mysql_port'); else  mysql_port=$(bashio::services "mysql" "port"); fi
if bashio::config.has_value "mysql_username"; then mysql_username=$(bashio::config 'mysql_username'); else mysql_username=$(bashio::services "mysql" "username") ; fi 
```

**Comment by frenck** [comment]
This request has come by a couple of times in the history of this add-on. The answer is the same: I have no intention of implementing this feature. If you need to set up an external database with this, this add-on is not for you.

../Frenck

**Comment by frenck** [comment]
MySQL support is pending removal for the next release of this add-on.

../Frenck
